cpp/src/io/parquet/page_enc.cu

```diff
@@ -1635,14 +1678,18 @@ __global__ void __launch_bounds__(1024) gpuGatherPages(EncColumnChunk *chunks, c
  * Similarly we merge up all the way till level 0 offsets
  */
 dremel_data get_dremel_data(column_view h_col,
-                            std::vector<bool> const &level_nullability,
+                            // TODO(cp): use device_span once it is converted to a single hd_vec
+                            rmm::device_uvector<uint8_t> const &d_nullability,
+                            std::vector<uint8_t> const &nullability,
                             rmm::cuda_stream_view stream)
 {
```

Comment:
I think it would be worth adding some comments to the docs for this function describing the unusual nature of the column_view being passed in (the "fragmented/parquet-i-fied" nature of it).

Reply:
Currently this is part of the constructor of `parquet_column_view`. We could move this around so that a `parquet_column_view` is given to it as an in/out parameter. It would take a pcv that doesn't have the rep/def level populated and calculate ans set it in. It wouldn't much change the execution.

The nature of a struct column being a single hierarchy now is meant to be understood elsewhere in the file too, (like in initFragments and encodePages).